lib/internal/bootstrap_node.js

```diff
@@ -327,9 +337,21 @@
           // nothing to be done about it at this point.
         }
 
-      // if we handled an error, then make sure any ticks get processed
       } else {
+        // If we handled an error, then make sure any ticks get processed
         NativeModule.require('timers').setImmediate(process._tickCallback);
+
+        // Emit the after() hooks now that the exception has been handled.
```

Comment:
* The comment is fine, but could you add a lot of details about why the after hooks needs to be emitted, and why we do this after `setImmediate()` is called.
* The `setImmediate()` is a new resource, so is that going to cause issues in the following cleanup?